auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java

```diff
@@ -27,7 +28,7 @@
 
   TypeAdapterFactory FACTORY = new TypeAdapterFactory() {
     private final Map<Class<?>, Constructor<? extends TypeAdapter>> adapters =
-        new ConcurrentHashMap<>();
+        Collections.synchronizedMap(new HashMap<Class<?>, Constructor<? extends TypeAdapter>>());
```

Comment:
btw, [LinkedHashMap is always better than HashMap.](https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/)

Reply:
Ordering doesn't matter here nor are we testing it, I'm not too worried about it :)

Reply:
Ordering _always_ matters regardless on whether you explicitly rely on it or not. That's the point of the post.

Reply:
I don't feel strongly about it in this case, but done in 24fc3b7